cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -138,28 +147,93 @@ func (pm *PipelineManager) Configure(ctx context.Context) (result *PipelineConfi
 		return result, fmt.Errorf("ensuring git remote: %w", err)
 	}
 
-	// *********** Create or update Azure Principal ***********
-	if pm.args.PipelineServicePrincipalName == "" {
-		// This format matches what the `az` cli uses when a name is not provided, with the prefix
-		// changed from "az-cli" to "az-dev"
-		pm.args.PipelineServicePrincipalName = fmt.Sprintf("az-dev-%s", time.Now().UTC().Format("01-02-2006-15-04-05"))
+	// Existing Service Principal Lookup strategy
+	// 1. --principal-id
+	// 2. --principal-name
```

Comment:
Shouldn't we fail if both principal-id and principal-name are added?

Reply:
Yes, I can add this check.

Reply:
Do we have a preference between allowing both, or maybe  an alternative implementation to overload `principal-name` to mean "name or id", treating as ID if guid?

Reply:
Having both available explicitly makes it super clear to the user what they can pass to each parameter.